src/SBInterpolatedImage.cpp

```diff
@@ -895,31 +881,61 @@ namespace galsim {
         }
         _flux = kValue(Position<double>(0.,0.)).real();
         dbg<<"flux = "<<_flux<<std::endl;
+        setCentroid();
+    }
+
+    void SBInterpolatedKImage::SBInterpolatedKImageImpl::setCentroid() const {
```

Comment:
There wasn't anything inappropriate.  I just thought this was more elegant than using the old `_cenIsSet` boolean member variable, more aligned with how SBII works, and simplifies `SBIKI.__getinitargs__`.  Calculating the centroid for an SBIKI is (surprisingly) only O(N), so less painful than computing the centroid of an SBII (O(N^2)), which is done in its constructor.